packages/hebao_v2/contracts/base/libwallet/QuotaLib.sol

```diff
@@ -98,9 +101,9 @@ library QuotaLib
         }
 
         Quota storage quota = wallet.quota;
-        quota.currentQuota = currentQuota(wallet).toUint128();
         quota.pendingQuota = newQuota.toUint128();
         quota.pendingUntil = effectiveTime.toUint64();
+        quota.currentQuota = currentQuota(wallet).toUint128();
```

Comment:
Sorry for asking this just now, I didn't think this change had any effect but on second thought it does change the behavior. 

It does seem like we first have to update `quota.currentQuota` in the case the current `quota.pendingQuota` is the actual current quota:  `q.pendingUntil <= block.timestamp ? q.pendingQuota : q.currentQuota`. So if with the current values `q.pendingUntil <= block.timestamp` is true than we have to first update `quota.currentQuota` because otherwise the correct current quota will be lost after the update.

For example:
- `currentQuota := 10`, `pendingQuota := 20` and `pendingUntil := 1000`.
- `block.timestamp := 2000` so `currentQuota(wallet) := 20`.
- `setQuota` is called with `newQuota := 50` and `effectiveTime := 4000`
- Correct behavior:  `currentQuota := 20`, `pendingQuota := 50`, and `pendingUntil := 4000`
- Behavior after this change: `currentQuota := 10`, `pendingQuota := 50`, and `pendingUntil := 4000`

Reply:
I did this change because when changeDailyQuotaWA is invoked, the currentQuota is expected to change immediately.
This issue is caused by the currentQuota is not updated when block.timestamp >= pendingUntil. So add quota.currentQuota = currentQuota(wallet).toUint128() at the begining of setQuota function can fix it.

Reply:
But the quota does change immediately no? `currentQuota()` will immediately return the new quota because `pendingUntil := 0` (`0 <= block.timestamp` so `pendingQuota` is the actual current quota). Is there a reason why you want the state variable `wallet.quota.currentQuota` to also be updated immediately? Like you said, the state variable for `currentQuota` is not updated elsewhere so we cannot depend on it being the actual current quota so we we still need to depend on the `currentQuota()` function everywhere.

I think adding the extra `quota.currentQuota = currentQuota(wallet).toUint128();` is okay, but I think the code works just as well without it so I don't really see the benefit of adding it.

Reply:
Makes sense